drivers/hmis/app/graphql/mutations/submit_household_assessments.rb

```diff
@@ -0,0 +1,105 @@
+module Mutations
+  class SubmitHouseholdAssessments < BaseMutation
+    description 'Submit multiple assessments in a household'
+
+    argument :assessment_ids, [ID], required: true
+    argument :confirmed, Boolean, 'Whether warnings have been confirmed', required: false
+
+    field :assessments, [Types::HmisSchema::Assessment], null: true
+
+    def resolve(assessment_ids:, confirmed:)
+      errors = HmisErrors::Errors.new
+
+      assessments = Hmis::Hud::Assessment.editable_by(current_user).
+        where(id: assessment_ids).
+        preload(:enrollment, :assessment_detail)
+
+      # Error: not all assessments found
+      if assessments.count != assessment_ids.size
+        errors.add :assessment, :not_found
+        return { errors: errors }
+      end
+
+      # Error: assessments do not all belong to the same household
+      household_ids = assessments.map { |a| a.enrollment.household_id }.uniq
+      if household_ids.count != 1
+        errors.add :assessment, :invalid, full_message: 'Assessments must all belong to the same household.'
+        return { errors: errors }
+      end
+
+      # Error: assessments do not have the same data collection stage
+      data_collection_stages = assessments.map { |a| a.assessment_detail.data_collection_stage }.uniq
+      if data_collection_stages.count != 1
+        errors.add :assessment, :invalid, full_message: 'Assessments must have the same data collection stage.'
+        return { errors: errors }
+      end
+
+      # If this includes an HoH Exit, check special restrictions
+      enrollments = assessments.map(&:enrollment)
+      includes_hoh = enrollments.map(&:relationship_to_ho_h).uniq.include?(1)
+      new_hoh_enrollment = nil
+      if assessments.first.exit? && includes_hoh
+        # FIXME: If exit dates can be in the future, `open_on_date` should check against HoH exit date
```

Comment:
Exit dates _should_ never be in the future, though I can see an argument for allowing it (HUD doesn't.)

Reply:
Ok. We don't restrict exit dates right now, but it is included in https://www.pivotaltracker.com/story/show/184407413